src/gilded_rose.js

```diff
@@ -21,45 +21,56 @@ const items = [
 
 updateQuality(items);
 */
-export function updateQuality(items) {
+export function updateQuality(itemsArray) {
+  let result = []
+  const items = [...itemsArray]
```

Comment:
I was trying to follow along with this but I wasn't entirely sure what's going on here. What's the reason for using the spread operator with the items array?

For example, let's say I have an array like this:
```
const myFavoriteColors = [ 'green', 'orange' ]
```
If I use the spread operator but also put it inside of an array, like what you're doing on line 26...
```
const myFavoriteColors = [ 'green', 'orange' ]
const myFavoriteColorsAlias = [ ... myFavoriteColors]
```
This is the same thing. I see that the output in your tests is different outputs in each, but I wasn't quite following what the purpose was.

Reply:
Yea I think I was starting to try and figure out how to do the whole immutable thing, I'll look more into how the code should be written to deal with updating the 'Item'